api/v1/db/oracledb/games-dao.js

```diff
@@ -0,0 +1,78 @@
+const appRoot = require('app-root-path');
+const _ = require('lodash');
+
+const { serializeGame, serializeGames } = require('../../serializers/games-serializer');
+
+const conn = appRoot.require('api/v1/db/oracledb/connection');
+
+/**
+ * @summary Return a list of games
+ * @function
+ * @returns {Promise<Object[]>} Promise object represents a list of games
+ */
+const getGames = async (queries) => {
+  // parse passed in parameters and construct query
+  const sqlParams = {
+    scoreMin: parseFloat(queries.scoreMin),
+    scoreMax: parseFloat(queries.scoreMax),
+  };
+  if (queries.name) {
+    sqlParams.name = queries.name;
+  }
+  if (queries.gameId) {
+    sqlParams.gameId = queries.gameId;
+  }
+  const sqlQuery = `
+    SELECT ID AS "id", DEVELOPER_ID AS "gameId", NAME AS "name", SCORE AS "score", RELEASE_DATE AS "releaseDate"
+    FROM VIDEO_GAMES
+    WHERE SCORE BETWEEN :scoreMin AND :scoreMax OR SCORE IS NULL
+    ${sqlParams.name ? 'AND NAME = :name' : ''}
+    ${sqlParams.gameId ? 'AND DEVELOPER_ID = :gameId' : ''}
+  `;
+
+  const connection = await conn.getConnection();
+  try {
+    // execute query and return results
+    const { rows } = await connection.execute(sqlQuery, sqlParams);
+    const serializedGames = serializeGames(rows, queries);
+    return serializedGames;
+  } finally {
+    connection.close();
+  }
+};
+
+/**
+ * @summary Return a specific game by unique ID
+ * @function
+ * @param {string} id Unique game ID
+ * @returns {Promise<Object>} Promise object represents a specific game or return undefined if
+ *                            term is not found
+ */
+const getGameById = async (id) => {
+  const connection = await conn.getConnection();
+  try {
+    const sqlParams = {
+      gameId: id,
+    };
+    const sqlQuery = `
+      SELECT ID AS "id", DEVELOPER_ID AS "gameId", NAME AS "name",
+      SCORE AS "score", RELEASE_DATE AS "releaseDate"
```

Comment:
The releaseDate format should follow the format you defined in openapi file here: https://github.com/osu-mist/game-curator-api/blob/master/openapi.yaml#L510. What I got currently is in this format: `"releaseDate": "28-OCT-08"`. You should make sure the raw data is serialized in your serializer.

Reply:
In addition, I tried to search the game with a score `GET /games/43`:

```json
{
    "links": {
        "self": "https://localhost/v1/games/43"
    },
    "data": {
        "type": "game",
        "id": "43",
        "links": {
            "self": "https://localhost/v1/games/43"
        },
        "attributes": {
            "name": "Path of Exile",
            "score": "2",
            "releaseDate": "15-OCT-13"
        }
    }
}
```

The score was parsed as string, but it should be a floating number: https://github.com/osu-mist/game-curator-api/blob/master/openapi.yaml#L501-L505

Reply:
Converted score and releaseDate into the correct formats.